modules/javafx.graphics/src/main/java/com/sun/prism/impl/BaseContext.java

```diff
@@ -108,6 +108,8 @@ public void flushVertexBuffer() {
     }
 
     protected final void flushMask() {
+        if (maskTex != null && maskTex.isSurfaceLost()) return;
```

Comment:
if maskText is null the subsequent calls could throw NPE, so maybe this should be:

```
        if (maskTex == null || maskTex.isSurfaceLost()) return;
```

Reply:
Fair point. I think this code works regardless because originally `maskTex` was supposed to be guarded by `curMaskRow` and `curMaskCol` below being set to zero.

I will change it to the above.

Reply:
I think they should remain unless we are absolutely certain that this is the only possibly place where `update` can be called on a texture that has been released.